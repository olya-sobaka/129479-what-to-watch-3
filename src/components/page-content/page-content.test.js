import React from "react";
import renderer from "react-test-renderer";
import PageContent from "./page-content";

const films = [
  {
    title: `Kiki's Delivery Service`,
    poster: `img/midnight-special.jpg`,
    id: 36,
    genre: `horror`
  },
  {
    title: `Kiki's Delivery Service`,
    poster: `img/midnight-special.jpg`,
    id: 37,
    genre: `drama`
  },
  {
    title: `Kiki's Delivery Service`,
    poster: `img/midnight-special.jpg`,
    id: 38,
    genre: `documentary`
  }
];

const mockFunc = () => {};

it(`PageContent renders correctly when prop genre is passed`, () => {

  const genre = `documentary`;

  const tree = renderer.create(
      <PageContent onGenreClick={mockFunc} onCardClick={mockFunc} films={films} genre={genre} />
  ).toJSON;

  expect(tree).toMatchSnapshot();
});

it(`PageContent renders correctly when prop moreLike is passed`, () => {

  const currentFilm = {
    id: 50,
    genre: `horror`
  };

  const tree = renderer.create(
      <PageContent onGenreClick={mockFunc} onCardClick={mockFunc} films={films} moreLike={true} currentFilm={currentFilm} />
  ).toJSON;

  expect(tree).toMatchSnapshot();
});
