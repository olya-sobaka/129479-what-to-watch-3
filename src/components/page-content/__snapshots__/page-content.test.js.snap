// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent renders correctly when prop genre is passed 1`] = `[Function]`;

exports[`PageContent renders correctly when prop moreLike is passed 1`] = `[Function]`;
